crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -208,7 +208,7 @@ impl AlgorithmUpdaterV1 {
         used: u64,
         capacity: NonZeroU64,
         block_bytes: u64,
-        _gas_price: u64,
+        fee_wei: u64,
```

Comment:
The fee that comes from the block is in Gwei

Reply:
Yeah. But we don't need to know that here. The caller can convert it for us. In the spirit of what you said:
> I think the algorithm should work with Wei, and all logic around gas price should work with Wei. We always can convert L2 gas price into Wei by using gas price and gas price factor.

It really doesn't matter either way (either wei :P), we just need to pick one and be clear to the caller which one it is.

Reply:
Then I think we need to introduce two new types `GWei` and `Wei` into our codebase and use it everywhere for fee and gas price(Even in state transition function).

In this case, we can be sure that we haven't forgotten to do the conversion.

It is a separate concern for a separate issue=) Could you create an issue please for it?

Reply:
Maybe? Fuel can be an L2 for other chains than Ethereum. I hesitated to put `wei` in the param name `fee_wei`, but I didn't know how else to specify--and it's much easier to change the one param name here than everywhere in our code. "This is just the Ethereum algorithm" 😬

Reply:
https://github.com/FuelLabs/fuel-core/issues/2248